src/testUtils/factories/pageEditorFactories.ts

```diff
@@ -74,38 +73,36 @@ const internalFormStateFactory = define<InternalFormStateOverride>({
     return [];
   },
   modMetadata: undefined,
-  type: StarterBrickTypes.SIDEBAR_PANEL,
+  permissions(): Permissions.Permissions {
+    return {
+      permissions: [],
+      origins: [],
+    };
+  },
   label: (i: number) => `Element ${i}`,
   modComponent: baseModComponentStateFactory,
-  // @ts-expect-error -- TODO: verify typings
-  starterBrick: derive<ModComponentFormState, StarterBrickDefinitionLike>(
-    ({ type }) => {
-      // FIXME: the starter brick type produced is not based on the type provided
-      const starterBrick = starterBrickDefinitionFactory();
-      if (type) {
-        starterBrick.definition.type = type;
-      }
-
-      return starterBrick;
-    },
-    "type",
-  ),
+  starterBrick: starterBrickDefinitionFactory,
 });
 
 export const formStateFactory = (
   override?: FactoryConfig<ModComponentFormState>,
   pipelineOverride?: BrickPipeline,
+  starterBrickOverride?: StarterBrickDefinitionLike,
 ): ModComponentFormState => {
```

Comment:
I'm not a fan of these overrides and all the `formStateFactory(undefined, undefined, { ... })` calls everywhere. 

```suggestion
export const formStateFactory = ({
  formStateOverride?: FactoryConfig<ModComponentFormState>,
  pipelineOverride?: BrickPipeline,
  starterBrickOverride?: StarterBrickDefinitionLike,
}): ModComponentFormState => {
```

Reply:
Yeah this is definitely better!

Reply:
Leaving this to a follow-up PR